packages/patrol_cli/lib/src/features/drive/drive_command.dart

```diff
@@ -248,34 +228,56 @@ class DriveCommand extends StagedCommand<DriveCommandConfig> {
     );
 
     var exitCode = 0;
-    for (final target in config.targets) {
-      _testRunner.addTest((device) async {
-        if (_disposeScope.disposed) {
-          _logger.fine('Skipping running $target...');
-          return;
+    _testRunner
+      ..builder = (target, device) async {
+        try {
+          await _flutterTool.build(target, device);
+        } catch (err) {
+          exitCode = 1;
+          _logger
+            ..severe(err)
+            ..severe(
+                'See the logs above to learn what happened. If the logs above '
```

Comment:
could you extract it somehow as I believe it will be used quite often?

Reply:
It's used only twice (rule of 3 doesn't apply 👅)

I plan to improve error reporting in the future (see #561).